src/module3/eventemitter-buffer-streams.ts

```diff
@@ -0,0 +1,68 @@
+import axios from "axios";
+import { EventEmitter, WithTime, exportCsvToTxt } from "./provided";
+
+export const run = () => {
+    task1();
+    task2().then(
+        () => task3()
+    );
+}
+
+const task1 = () => {
+    console.log('Task 1: Event Emitter');
+
+    const emitter = new EventEmitter();
+
+    // Test addListener and emit
+    emitter.addListener('testEvent', (data) => console.log(data));
+    emitter.emit('testEvent', 'Hello, World!'); // Should log 'Hello, World!'
+
+    // Test listenerCount
+    console.log(emitter.listenerCount('testEvent')); // Should log 1
+
+    // Test rawListeners
+    console.log(emitter.rawListeners('testEvent')); // Should log [ [Function] ]
+
+    // Test removeListener
+    emitter.removeListener('testEvent', console.log);
+    console.log(emitter.listenerCount('testEvent')); // Should log 0

```

Comment:
It logs 1 instead of 0.
If you create a new fn and update this consumer code like below, it works:

`const log = (data: any) => console.log(data);`
`emitter.addListener('testEvent',_ log);`
...
`emitter.removeListener('testEvent', log);`
 
 Guess it's a fault in the exampler and not your code.

Reply:
Ohh, that makes sense

Reply:
Up to you, I will give full score anyway. :)